### PR TITLE
feat: add browseWithoutSelecting prop

### LINKED ISSUE #15: Do not change current selected date when changing month
First of all, let me thank you for a great datepicker.

I've noticed a behavior that seems to me unusual. When the month or year is switched, the new date is immediately selected.

For example, May 5 is currently selected. If we move the calendar forward one month, June 5 will be selected immediately, and so on. In my opinion the date should not be selected unless I explicitly click on the new date in the new month. At least this is how the datepicker works in Google calendar or the classic jQuery UI [datepicker](https://jqueryui.com/datepicker/).

Is it possible to change the behavior of your datepicker, or at least make a setting for this behavior?

### PR DESCRIPTION
**Pull request by e0**
This is aimed to resolve #15, so that when the year or month is changed, the date input does not immediately change. In addition the date input also does not change when the user uses the arrow keys in the date picker. With this change, the only time when the date picker would update the date input is when the user clicks on a date or presses the `Enter` key, given that the new `delayUpdate` property is set to `true` (`false` by default so it does not interfere with the current functionality).

@probablykasper please let me know if this is something you are interested in merging in. I'm happy to make additional changes.

### CODE REVIEW

**Comment by probablykasper** [comment]
Thanks for the PR, appreciate it!

I think the implementation here could be improved. When you go to a new month/year, it still highlights a date in that month, which makes it look like that date is selected even though it isn't. I also think this property should exist in `DatePicker`, not just `DateInput`.

I'd change the name `delayUpdate`, though it's a bit hard to think of a good name. Maybe `browseWithoutSelecting`?

**Comment by e0** [comment]
Thanks for the feedback @probablykasper! Nice that you spotted the problem with the date still being highlighted. I like the name change suggestion as well. Will try to fix these now.

**Comment by e0** [comment]
I have made some updates to the `DatePicker` now. Not sure about the implementation and if you want to see anything changed, please let me know.

In terms of functionality, no calendar days should be highlighted now when changing month/year. The exception is when changing via the `keydown` events since it feels the most intuitive, for example when pressing the `Left` key when the currently highlighted date is `2022-01-01`.

**Comment by probablykasper** [comment]
Nice, that seems good. It's a bit bugged though, the `value` still updates sometimes when you switch month/year

**Comment by e0** [comment]
I was wondering if that would be the correct behavior. For example, if the year is changed using the select, I thought it would be more straightforward to still change the `value` but not highlight it so that when pressing one of the arrow keys would select a date in the currently displayed month. I think I was still ignoring the fact that the `DatePicker` can used as a standalone component without the `DateInput` though. I will try to have another look at this. Thanks again!

**Comment by probablykasper** [comment]
The `value` is what you actually use for example if you send a form, so it has to be in sync with what the `DateInput` shows

**Comment by e0** [comment]
That makes a lot of sense. I added a guard in the `updateShownDate` function now to conditionally update the `value`. I'm not too sure if I understood the purpose of `shownDate` so please let me know if there is some unwanted side-effects from this.

File: src/lib/DatePicker.svelte [hunk 1]
```diff
@@ -29,7 +29,7 @@
   /** Update the shownDate. The date is only selected if a date is already selected */
   function updateShownDate(updater: (date: Date) => Date) {
     shownDate = updater(new Date(shownDate.getTime()))
-    if (value && shownDate.getTime() !== value.getTime()) {
+    if (!browseWithoutSelecting && value && shownDate.getTime() !== value.getTime()) {
```
Comment (probablykasper) [inline src/lib/DatePicker.svelte:32]: This has some unintended consequences.

`updateShownDate` is used to clamp the value within the `min` and `max` dates. So if you have `browseWithoutSelecting`, and you update the `min` date, you can now end up with a date that's outside the desired range.

It also makes it so selecting dates with the arrow keys doesn't work with `browseWithoutSelecting` (could be other things too, I haven't checked everything)

**Review by probablykasper — CHANGES_REQUESTED** [review]

**Comment by probablykasper** [comment]
Could we maybe just use `shownDate` instead of having `highlightedDate`?

File: src/lib/DateInput.svelte [hunk 2]
```diff
@@ -134,6 +136,7 @@
     if (closeOnSelection) {
       visible = false
     }
+    innerStore.set(value)
```
[the remark this answers is not in the document]
Comment (e0) [reply src/lib/DateInput.svelte]: Absolutely. This should have been removed in an earlier commit after the logic was added in the `DatePicker`.

**Comment by e0** [comment]
> Could we maybe just use `shownDate` instead of having `highlightedDate`?

Thanks for the review. Simply using `shownDate` directly does not work from what I tried as it still changes for example when browsing with arrow keys. I think it is possible to rewrite things so that only `shownDate` is used. I think I need to have a more thorough understanding of things to make sure that I don't break anything (for example as you mentioned [here](https://github.com/probablykasper/date-picker-svelte/pull/18#discussion_r928473763)). I'm not sure when I would have the time to do this though. Would you rather that I close the PR for now and open it up again in the future?

**Comment by probablykasper** [comment]
I tried my hand at it in PR #18

**Pull request merged**